handlers/authorization/check_student_existance.py

```diff
@@ -9,39 +9,46 @@
 import os
 
 from fastapi import APIRouter
-from fastapi.responses import JSONResponse
+from fastapi import status as http_status
 
+from schemas.jsend import JSENDOutSchema
+from components.exceptions import BackendException
 
 router = APIRouter()
 
 
-@router.post("/check-student-existance", response_model=StudentCheckExistanceOut, tags=["Authorization"])
+@router.post("/check-student-existance", response_model=JSENDOutSchema[StudentCheckExistanceOut],
+             tags=["Authorization"])  # TODO syntax error, there is need to check path in other modules
 async def check_student(student: StudentCheckExistanceIn):
-
-    query = student_table.select().where(student_table.c.full_name == student.full_name, 
-                                    student_table.c.telephone_number == student.telephone_number)
+    query = student_table.select().where(student_table.c.full_name == student.full_name,
+                                         student_table.c.telephone_number == student.telephone_number)
     result = await database.fetch_one(query)
 
     if not result:
-        return JSONResponse(status_code=404, content={"message": "Дані про студента не знайдено. " \
-                                                                "Будь ласка, спробуйте ще раз."})
+        raise BackendException(
+            message="There is no students data. Please, try again.",
+            code=http_status.HTTP_404_NOT_FOUND
+        )
 
     student_id = result.student_id
 
     token = hashlib.sha1(os.urandom(128)).hexdigest()
     expires = datetime.utcnow() + timedelta(seconds=Settings.TOKEN_LIFE_TIME)
 
     query = one_time_token.insert().values(student_id=student_id, token=token,
-                                           expires=expires).returning(one_time_token.c.token_id)                      
+                                           expires=expires).returning(one_time_token.c.token_id)
     last_record_id = await database.execute(query)
 
     query = one_time_token.select().where(one_time_token.c.token_id == last_record_id)
     result = await database.fetch_one(query)
 
     response = {
-                'token': result.token,
-                'student': result.student_id, 
-                'expires': result.expires
+        'token': result.token,
+        'student': result.student_id,
+        'expires': result.expires
     }
 
-    return response
+    return JSENDOutSchema[StudentCheckExistanceOut](
+        data=response,
```

Comment:
is't ok that you use wrapper JSENDOutSchema[StudentCheckExistanceOut] 2 times - there is and in response_model variable?

Reply:
This is robust, because it provides benefits of static type checking. E. g. it prevents programmer from making typos before they even run the code:
```python
Schema(
    datta=...,  # TypeError: Schema.__init__() got an unexpected keyword argument 'datta'
    mesage=...,
)
```

Usually, if you declare function with type hints

```python
def my_func(x: int, y: str) -> SomeClass:
    ...
    return SomeClass(...)
```
there is no way to avoid writing `SomeClass` twice. This is similar situation, because `response_model` parameter has similar meaning to return type.

Reply:
Note, I was talking about **old approach** being robust. Second part of my comment explains why writing `JSENDOutSchema[StudentCheckExistanceOut]` twice is acceptable